app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -387,12 +396,21 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         navHost.navController.navigate(action)
     }
 
+    private fun Toolbar.setChildrenClickable(boolean: Boolean, delayTime: Long) {
+        Timer().schedule(delayTime) {
+            this@setChildrenClickable.children.forEach {
+                it.isClickable = boolean
+            }
+        }
```

Comment:
Hm. Couldn't this cause a short timed leak of the activity? This is a bit scary since two `HomeActivity` existing at the same time, trying to render the same session, would cause a crash. And I think things like theme switches will recreate the activity immediately?

Reply:
Yeah, but I don't see any scenario where both theme switch and back button click would happen in the same time.